site/_posts/2021-01-13-why-outsource-auth.md

```diff
@@ -0,0 +1,84 @@
+
+Properly storing passwords is an incredibly low bar, and yet companies that manage their own auth still do it incorrectly all the time.
+
+## Consequences of an auth outage
+
+While less damaging than breaches, outages can still cause reputational damage and liability issues if your SLAs make uptime guarantees. Similarly to breaches, if you outsource your auth system it's likely that any auth outage will extend beyond your company. 
+
+As an example, when Microsoft's Azure Active Directory (AAD) [went down for a good portion of the afternoon late last year](https://www.zdnet.com/article/microsofts-azure-ad-authentication-outage-what-went-wrong/), logins for applications across the internet stopped working. 
+
+When your competitors' authorization systems are down at the same time yours are, nobody blames you for it, but when you're the only company having issues, you suffer reputationally. No matter what your outsourced auth system is (FusionAuth, Cognito, AAD, etc), you can be almost certain that you won't be alone in the event of an outage.
+
+## Maintainability
+
+There are trade-offs here. The benefits to an in-house system include:
+
+* Your engineers can design the exact system to fit your needs, and you'll have unrestricted ability to add very specific features if requirements change over time.
+* If the same engineers who built the system are maintaining it, then they'll have the context required to anticipate issues as they add features.
+
+However, the drawbacks can be large:
+
+* Complex new features can take significant time to build. Outsourced auth systems likely have these features already built (things like multi-factor authentication, user management interfaces, analytics and audit logs, and brute force hack detection, among others).
+* With an in-house solution, you'll have to budget time to monitor new security threats and patch your system in a rapidly evolving threat landscape.
+* You won't get the benefits of a dedicated team that are constantly improving your auth system. This is actually a bigger deal than it seems, because if you outsource auth then other companies will also be filing issue reports and feature requests on your behalf, so you reap the benefits of those extra eyes as well.
+* In the case of mergers or acquisitions, an in-house solution is likely to be terrible at combining different databases of users and managing things like duplicates and incomplete data. Enterprise Identity Unification (EIU) efforts can founder on internal auth systems.
```

Comment:
We might need a link to something that describes EIU here. I think the last sentence isn't correct either. Should it be "flounder"? This entire bullet could use some work.